lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+    if (typeof matches[0].displayName === 'string' &&
+    matches[0].displayName.toLowerCase().indexOf(searchText.toLowerCase()) === 0) {
+      setPrevSearchText(searchText);
+      onSearchTextChange(matches[0].displayName);
+    }
+  }
+
+  // Clamp or nullify focusableBtnIndex whenever the number of matches changes
+  useEffect(function() {
+    if (matches.length) {
+      setFocusableBtnIndex(clamp(0, matches.length - 1, focusableBtnIndex ?? 0));
+      if (autoComplete) {
+        setInlineOption();
+      }
+    }
+    else {
+      setFocusableBtnIndex(null);
+    }
+  }, [matches]);
+
+  // If the dropdown is closed and use arrowkeys to navigate back to the items in the dropdown menu,
+  // the ref will fire before state is updated, this will insure the items have visual focus and it's inview.
+  useEffect(function() {
+    if (elToFocusId && typeof focusableBtnIndex === 'number') {
+      const el = dropdownRef.current?.children[focusableBtnIndex] as HTMLElement | null;
+      if (el) {
+        handleElInView(el);
+      }
+    }
+  }, [elToFocusId]);
+
+  useEffect(function() {
+    if (prevSearchText && autoComplete) {
+      inputRef.current?.querySelector('input')?.setSelectionRange(prevSearchText.length, searchText.length);
+      focusFirst();
+    }
+  }, [prevSearchText]);
+
+  /*
+   * Horrible Hack: When an element within the dropdown is removed from the DOM while it is focused, we want
+   * to move focus to the text input.  It turns out that this is very difficult to track in React, since
+   * useEffect and useLayoutEffect generally fire too late - after the element has already been removed and
+   * lost whatever focus it might've had. The only other way to get this info is with a useLayoutEffect handler
+   * _in the component that was unmounted_, e.g. in NxButton. That would require adding new props for a special use
+   * case to not only NxButton, but also NxLoadWrapper and NxLoadError. Just querying who has focus on every render
+   * seemed like the less bad option.
+   */
+  elFocusedOnMostRecentRender.current = typeof document === 'undefined' ? null : document.activeElement;
+
+  const checkForRemovedFocusedEl = useCallback(function checkForRemovedFocusedEl(mutations: MutationRecord[]) {
+    const nodeContainedFocus = (el: Node) => el.contains(elFocusedOnMostRecentRender.current),
+        nodeListContainedFocus =
+            pipe<[NodeList], Node[], boolean>(Array.from, any(nodeContainedFocus)),
+        focusedChildWasRemoved =
+            any(pipe(prop('removedNodes'), nodeListContainedFocus), mutations);
+
+    if (focusedChildWasRemoved) {
+      focusTextInput();
+    }
+  }, []);
+
+  useMutationObserver(dropdownRef, checkForRemovedFocusedEl, { childList: true });
+
+  function handleOnClick(match: DataItem<T>) {
+    onSelect(match);
+    focusTextInput();
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleElInView(el: HTMLElement) {
```

Comment:
This is one thing that using the roving tabindex approach might make easier - the browser automatically scrolls focused elements into view

Reply:
It has been updated here ea9ebf82532aa07b26529b0fb749c5886569f3f5.